src/reports/suite.js

```diff
@@ -0,0 +1,73 @@
+export default class SuiteReport {
+	constructor( name, parentSuite ) {
+		this.name = name;
+		this.fullName = parentSuite ? parentSuite.fullName.concat( name ) : [];
+
+		this.tests = [];
+		this.childSuites = [];
+
+		if ( parentSuite ) {
+			parentSuite.pushChildSuite( this );
+		}
+	}
+
+	start( recordTime ) {
+		if ( recordTime ) {
+			this._startTime = Date.now();
+		}
+
+		return {
+			name: this.name,
+			fullName: this.fullName,
```

Comment:
This should probably expose a copy, to prevent callbacks from manipulating private data.